### ISSUE TITLE
FList can't be pickled

### ISSUE THREAD

**Issue by ismael-mendoza** [aboucaud/galcheat#88]
As I'm trying to replace the survey functionality of BTK with galcheat, I realized that I can't pickle the galcheat.survey.Survey class due to it containing a dynamically generated dataclass `FList`. See [here](https://github.com/uqfoundation/dill/issues/56) and [here](https://stackoverflow.com/questions/65382074/pickling-an-instance-of-a-dataclass). 

Would it be possible to replace it with just a dictionary? Or why is the FList necessary? Thanks you! The reason I need to pickle it is to run multiprocessing which internally pickles objects. 

One workaround that was suggested was to do `FList.__module__ = __name__` (where it is created). But it seems a bit hacky and perhaps easier to replace with a dictionary if possible 

@aboucaud 

```
_pickle.PicklingError: Can't pickle <class 'types.LSSTFilterList'>: attribute lookup LSSTFilterList on types failed
```

**Comment by aboucaud** [comment]
The reason I created these dynamical lists was to enable the possibility of writing 
```python
LSST.filters.u.effective_wavelength
```
and having autocompletion in ipython or any IDE, instead of writing
```python
LSST.get_filter('u').effective_wavelength
```
where you actually need to check first which filters are available.

**Comment by aboucaud** [comment]
It cannot be solved through the trick of writing `FList.__module__ = __name__` because `FList` is just a placeholder for the dataclass created specifically for each survey given its filters : `LSSTFilterList` for `LSST`, `DESFilterList` for `DES`, etc. 

The only option seems to abandon that functionality for a dictionary, but sadly that was one of the coolest features I had created for galcheat 😢

**Comment by ismael-mendoza** [comment]
Thanks @aboucaud and @thuiop for looking into this. Thanks @aboucaud for explaining your reasoning on doing this. It is indeed unfortunate that we need to abandon the FList due to multiprocessing solely and that other workarounds don't seem to work. 

I'll think more about this today and let you know if I can come up with a workaround in BTK.

**Comment by aboucaud** [comment]
Closed by #92
